From aecb6f76c711320ab096bba37af80f46140d5564 Mon Sep 17 00:00:00 2001
From: bergner <bergner@138bc75d-0d04-0410-961f-82ee72b054a4>
Date: Thu, 22 May 2014 16:07:07 +0000
Subject: [PATCH 003/134] 	Backport from mainline 	2014-05-22  Peter
 Bergner  <bergner@vnet.ibm.com>

gcc/
	* config/rs6000/htm.md (ttest): Use correct shift value to get CR0.

gcc/testsuite/
	* gcc.target/powerpc/htm-ttest.c: New test.


git-svn-id: svn+ssh://gcc.gnu.org/svn/gcc/branches/gcc-4_8-branch@210818 138bc75d-0d04-0410-961f-82ee72b054a4
---
 gcc/ChangeLog                                |  7 +++++++
 gcc/config/rs6000/htm.md                     |  2 +-
 gcc/testsuite/ChangeLog                      |  7 +++++++
 gcc/testsuite/gcc.target/powerpc/htm-ttest.c | 14 ++++++++++++++
 4 files changed, 29 insertions(+), 1 deletion(-)
 create mode 100644 gcc/testsuite/gcc.target/powerpc/htm-ttest.c

diff --git a/gcc/ChangeLog b/gcc/ChangeLog
index 97346e8..667a495 100644
--- a/gcc/ChangeLog
+++ b/gcc/ChangeLog
@@ -1,3 +1,10 @@
+2014-05-22  Peter Bergner  <bergner@vnet.ibm.com>
+
+	Backport from mainline
+	2014-05-22  Peter Bergner  <bergner@vnet.ibm.com>
+
+	* config/rs6000/htm.md (ttest): Use correct shift value to get CR0.
+
 2014-05-22  Richard Earnshaw  <rearnsha@arm.com>
 
 	PR target/61208
diff --git a/gcc/config/rs6000/htm.md b/gcc/config/rs6000/htm.md
index e8ec91a..1a1e435 100644
--- a/gcc/config/rs6000/htm.md
+++ b/gcc/config/rs6000/htm.md
@@ -179,7 +179,7 @@
 			     (const_int 0)]
 			    UNSPECV_HTM_TABORTWCI))
    (set (subreg:CC (match_dup 2) 0) (match_dup 1))
-   (set (match_dup 3) (lshiftrt:SI (match_dup 2) (const_int 24)))
+   (set (match_dup 3) (lshiftrt:SI (match_dup 2) (const_int 28)))
    (parallel [(set (match_operand:SI 0 "int_reg_operand" "")
 		   (and:SI (match_dup 3) (const_int 15)))
               (clobber (scratch:CC))])]
diff --git a/gcc/testsuite/ChangeLog b/gcc/testsuite/ChangeLog
index 28598e5..1d4513f 100644
--- a/gcc/testsuite/ChangeLog
+++ b/gcc/testsuite/ChangeLog
@@ -1,3 +1,10 @@
+2014-05-22  Peter Bergner  <bergner@vnet.ibm.com>
+
+	Backport from mainline
+	2014-05-22  Peter Bergner  <bergner@vnet.ibm.com>
+
+	* gcc.target/powerpc/htm-ttest.c: New test.
+
 2014-05-22  Release Manager
 
 	* GCC 4.8.3 released.
diff --git a/gcc/testsuite/gcc.target/powerpc/htm-ttest.c b/gcc/testsuite/gcc.target/powerpc/htm-ttest.c
new file mode 100644
index 0000000..29cbd5b
--- /dev/null
+++ b/gcc/testsuite/gcc.target/powerpc/htm-ttest.c
@@ -0,0 +1,14 @@
+/* { dg-do compile { target { powerpc*-*-* } } } */
+/* { dg-skip-if "" { powerpc*-*-darwin* } { "*" } { "" } } */
+/* { dg-require-effective-target powerpc_htm_ok } */
+/* { dg-options "-O2 -mhtm" } */
+
+/* { dg-final { scan-assembler "rlwinm r?\[0-9\]+,r?\[0-9\]+,3,30,31" { target { ilp32 } } } } */
+/* { dg-final { scan-assembler "rldicl r?\[0-9\]+,r?\[0-9\]+,35,62" { target { lp64 } } } } */
+
+#include <htmintrin.h>
+long
+ttest (void)
+{
+  return _HTM_STATE(__builtin_ttest());
+}
-- 
1.8.5.5

